### PR TITLE
[Fix] Avoid pickling entire LLMEngine for Ray workers

### PR DESCRIPTION
**Pull request by njhill**
`self` references have been inadvertently introduced into the closure used to initialize the Ray workers, which is causing the whole `LLMEngine` object to be pickled with it.

This fixes it to reference the `lora_config` and `kv_cache_dtype` directly.

### CODE REVIEW

**Review by Yard1 — APPROVED** [review]
Looks good, can you add a short description?

**Comment by simon-mo** [comment]
You can implement the `__reduce__` method in `LLMEngine` and throw an error in there so this won't happen again in the future.

**Comment by njhill** [comment]
Good idea @simon-mo, now added.

File: vllm/engine/llm_engine.py [hunk 1]
```diff
@@ -158,6 +158,11 @@ def __init__(
         if USE_RAY_COMPILED_DAG:
             self.forward_dag = self._compiled_ray_dag()
 
+    def __reduce__(self):
+        # This is to ensure that the LLMEngine is not referenced in
+        # the closure used to initialize Ray worker actors
+        raise Exception("LLMEngine should not be pickled!")
```
Comment (Yard1) [inline vllm/engine/llm_engine.py:164]:
```suggestion
        raise RuntimeException("LLMEngine should not be pickled!")
```

File: vllm/engine/llm_engine.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ def __init__(
     def __reduce__(self):
         # This is to ensure that the LLMEngine is not referenced in
         # the closure used to initialize Ray worker actors
-        raise Exception("LLMEngine should not be pickled!")
+        raise RuntimeException("LLMEngine should not be pickled!")
```
Comment (Yard1) [inline vllm/engine/llm_engine.py:164]:
```suggestion
        raise RuntimeError("LLMEngine should not be pickled!")
```

**Pull request merged**